woocommerce/woocommerce
# dev: bumped typescript to 5.7.2

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

- Upgrades Typescript to 5.7.2 across the monorepo to gain access to [nocheck](https://devblogs.microsoft.com/typescript/announcing-typescript-5-6/#the---nocheck-option) and [Allow --build with Intermediate Errors](https://devblogs.microsoft.com/typescript/announcing-typescript-5-6/#allow---build-with-intermediate-errors) 
- Bumped `@types/nodes` as it was causing an error on type mismatch of AbortSignal after the TS upgrade
- Fixed some new type errors in 
  - packages/js/data/src/import/selectors.ts
  - packages/js/customer-effort-score/src/components/customer-effort-score-tracks-container/index.js
  - packages/js/customer-effort-score/src/components/customer-effort-score-tracks/index.js



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. WC should build fine, and an inspection of the type error fixes should be sufficiently convincing that there will be no effective change to the code



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Upgraded Typescript in the monorepo to 5.7.2
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
